### PR TITLE
[Merged by Bors] - feat: add logLintIf

### PR DESCRIPTION
**Pull request by fpvandoorn**
* This logs a linter error only if the option is set
* I thought that was the behavior of `logLint` already, but that was false. This still seems to be a good option to have.
* Part of the reason I was mislead is that `Lean.Linter.logLint` (and `Lean.logWarningAt`) have an unused `MonadOptions` argument

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by fpvandoorn** [comment]
Oops, I indeed did.

File: Mathlib/Lean/Linter.lean [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+/-!
+# Additional declarations related to linters
+-/
+
+open Lean Elab
+
+namespace Lean.Linter
+
+def logLint? [Monad m] [MonadLog m] [AddMessageContext m] [MonadOptions m]
```
Comment (gebner) [inline Mathlib/Lean/Linter.lean:16]:
```suggestion
def logLintIf [Monad m] [MonadLog m] [AddMessageContext m] [MonadOptions m]
```
The `?` suffix is typically used to indicate an `Option` in the return type.

**Comment by gebner** [comment]
bors d+

File: Mathlib/Lean/Linter.lean (hunk 1 above)
Comment (fpvandoorn) [reply Mathlib/Lean/Linter.lean:16]: You're right, this is different

File: Mathlib/Lean/Linter.lean [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+/-!
+# Additional declarations related to linters
+-/
+
+open Lean Elab
+
+namespace Lean.Linter
+
+def LogLintIf [Monad m] [MonadLog m] [AddMessageContext m] [MonadOptions m]
```
Comment (semorrison) [inline Mathlib/Lean/Linter.lean:16]: Missing doc-string.

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**